test/e2e/pkg/testnet.go

```diff
@@ -78,6 +78,7 @@ type Node struct {
 	Seeds            []*Node
 	PersistentPeers  []*Node
 	Perturbations    []Perturbation
+	Misbehaviors     map[string]string
```

Comment:
This should be `map[int64]string`. In the manifest, we have to use `string` keys due to TOML semantics, but in the canonical testnet representation we should use appropriate types.

Reply:
There was a reason I did this. I think it was something along the lines of being annoyed at having to convert from `map[string]string` to `map[int64]string` to `string` if it was separate and `map[int64]cs.Behavior`

Reply:
Yeah, but e.g. in tests we'll always have to convert these from strings to ints everywhere they're used. It's better to convert it once into a proper format than to spread it all over.

Reply:
Ah I remember the other thing which is that once converting it from `map[string]string` to `map[int64]string` I then have to convert it back to `map[string]string` for the app config which is written in toml

Reply:
Yeah, I know, it's a bit of a hassle. But the canonical data model should use proper types, and whatever gymnastics we have to do when serializing/deserializing to e.g. TOML is just a penalty we'll have to pay. This way we'll only convert a fixed number of times (when serializing/deserializing) instead of every time we want to actually use the data.